Allow (mobile) plugins to define telemetry logs.

dogmaphobic opened the pull request.

DonLakeFlyer commented on src/QGCApplication.cc:
```diff
@@ -503,31 +503,39 @@ void QGCApplication::criticalMessageBoxOnMainThread(const QString& title, const
-        }
-        QString saveFilePath = saveDir.absoluteFilePath(saveFileName);
+    // Are we supposed to log?
+    if(toolbox()->settingsManager()->appSettings()->telemetrySave()->rawValue().toBool()) {
```
You shouldn't need this. You should never get here if that isn't set.

DonLakeFlyer replied:
The only case would be if you crash and there is a leftover log file. But if that happens then save was on at the time you crashed. Hence you would want to save that anyway.

DonLakeFlyer replied:
Ah, I hate the stupid review thing...

dogmaphobic commented:
I don't know what I did but your notes, along with my answers disappeared. I guess I see why you hate these code reviews :)
Either way, yes. I had added that before understanding the *Always Write To Temp* deal. I've removed it.
